src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js

```diff
@@ -21,27 +22,15 @@ import {
   Tr,
 } from '@patternfly/react-table';
 
-import api from '../../../api';
+import { useGetActivationKeyInformationQuery } from '../../../store/apiSlice';
 
 const ActivationKeyInformation = () => {
   const { getState } = useFormApi();
 
   const activationKey = getState()?.values?.['subscription-activation-key'];
-  const [role, setRole] = useState(undefined);
-  const [serviceLevel, setServiceLevel] = useState(undefined);
-  const [usage, setUsage] = useState(undefined);
-  const [additionalRepositories, setRepositories] = useState(undefined);
 
-  useEffect(() => {
-    const fetchKeyInformation = async () => {
-      const data = await api.getActivationKey(activationKey);
-      setRole(data?.role);
-      setServiceLevel(data?.serviceLevel);
-      setUsage(data?.usage);
-      setRepositories(data?.additionalRepositories);
-    };
-    fetchKeyInformation();
-  }, []);
+  const { data: activationKeyInfo, isSuccess: isSuccessActivationKeyInfo } =
+    useGetActivationKeyInformationQuery(activationKey);
```

Comment:
activationKey could theoretically be `undefined`, and in that case our query would return an error. `isSuccess` would remain false, and we would just show spinners indefinitely.

We might want to also extract `isError` and handle the error state separately (not show the spinner, instead display an error message). We could choose whether or not to render the spinners based on `isFetching`. 

We could also use the query's skip option to skip the query entirely if activationKey === undefined:
https://redux-toolkit.js.org/rtk-query/usage/conditional-fetching

But I still think it would be good to handle the error state properly. What if, for instance, the key has been deleted? It would be good to notify the user in that case.

Reply:
Great point, added error handling!

![image](https://user-images.githubusercontent.com/49452678/231992612-a999c91d-326e-4e8b-a3e1-ec9039b2527d.png)